### ISSUE TITLE
Issue with generating trees in PhyML

### ISSUE THREAD

**Issue by djh23887** [simonhmartin/twisst#37]
Hello,

I'm working on conducting a Twisst analysis, but I'm running into issues creating sliding window trees in PhyML. I've been trying to run the phyml_sliding_windows.py script as follows (after running parseVCF.py):

`python phyml_sliding_windows.py -g Mac2Renumberfortwisst.geno.gz --prefix output.phyml_bioml.w50 -w 50 --windType sites --model GTR --phyml /programs/phyml/bin/phyml`

This outputs .trees and .data files, but the .trees file is much smaller than I think it should be (<100 B).  When I feed that file into twisst.py, it seems that all the trees result in a "Warning - failed to read tree" message.  Do you have any idea what could be causing the issue with PhyML or if there's an obvious issue in my command?

Thanks for your help.

Best,
Declan

**Comment by simonhmartin** [comment]
Hi Declan,
There are various potential problems. I recommend running visualising the output files in a text viewer or in the terminal to see if they contain data. The window data file also contains infomration about the number of sites per window, and whether a tree was generated for each window (there should be a likelihood value provided if there was a tree). You can run this script with the `--test` option, which will just attempt 10 windows, and preserve the temporary alignment files, so you can also look at those. You can also add `--log phyml_log.txt`, which will save all the messages from phyml, so you can see if it is running into problems.

In summary, you will need to explore a bit to figure out what your specific problem is. Most often it is a failure to produce windows due to too much missing data. This can sometimes be solved with the `--Mi` option.

**Comment by djh23887** [comment]
Hi Simon,

Thanks for getting back to me.  After some looking and experimenting, it seems that missing data was the issue as adding `--Mi 10` fixed the problem.

Thanks for your help!

Best,
Declan

**Comment by simonhmartin** [comment]
Great. No problem.
